topic-operator/src/main/java/io/strimzi/operator/topic/v2/BatchingTopicController.java

```diff
@@ -416,9 +422,11 @@ private void updateInternal(List<ReconcilableTopic> topics) {
             deleteInternal(toBeDeleted, false);
         }
 
+        var remainingAfterDeletions = partitionedByDeletion.get(false);
+        remainingAfterDeletions.forEach(rt -> rt.startTimer(metrics));
```

Comment:
Why is it correct that we start the timer when processing the batch, rather than from the point we're told about the change to the topic? (i.e. why are we excluding the time in the queue?)

Reply:
The User Operator also has an event queue, and we are excluding time spent there for this metric. We could add a `strimzi.queue.duration` timer that could reveal a pathological situation where lots of changes are done on the same KafkaTopic, starving other resources. That said, we already have the cheaper `strimzi.reconciliations.locked` counter for that. Wdyt?

Reply:
If we're being consistent with UO then I guess I'm fine with keeping it like this.